pkg/controllers/gateway/params_test.go

```diff
@@ -55,6 +55,97 @@ func TestGetParams(t *testing.T) {
 				}),
 			),
 		},
+		{
+			name: "ConfigMap found. Uses default params on missing ref",
+			gatewayClass: &gatewayv1beta1.GatewayClass{
+				Spec: gatewayv1beta1.GatewayClassSpec{
+					ParametersRef: nil,
+				},
+			},
+			paramsObj: &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-params",
+					Namespace: "test-ns",
+				},
+				Data: map[string]string{
+					"params": `{"downstreamClass": "istio"}`,
+				},
+			},
+			assertParams: and(
+				noError,
+				paramsEqual(Params{
+					DownstreamClass: "istio",
+				}),
+			),
+		},
+		{
+			name: "Misconfigured ConfigMap",
+			gatewayClass: &gatewayv1beta1.GatewayClass{
+				Spec: gatewayv1beta1.GatewayClassSpec{
+					ParametersRef: &gatewayv1beta1.ParametersReference{
+						Group:     "",
+						Kind:      "ConfigMap",
+						Name:      "test-params",
+						Namespace: addr(gatewayv1beta1.Namespace("test-ns")),
```

Comment:
I think you can get rid of the addr function and use testutil.Pointer everywhere?

Reply:
missed one! I knew those sneaky calls would get me. Done